nft_ingester/src/metadata_workers/json_worker.rs

```diff
@@ -0,0 +1,374 @@
+                            storage_mutability: mutability.clone(),
+                        },
+                    );
+                    pg_updates.push(UpdatedTask {
+                        status: TaskStatus::Success,
+                        etag,
+                        last_modified_at,
+                        metadata_url: metadata_url.clone(),
+                        mutability: mutability.to_string(),
+                    });
+
+                    self.metrics.inc_tasks("media", MetricStatus::SUCCESS);
+                },
+                Err(json_err) => match json_err {
+                    JsonDownloaderError::GotNotJsonFile => {
+                        pg_updates.push(UpdatedTask {
+                            status: TaskStatus::Failed,
+                            metadata_url: metadata_url.clone(),
+                            etag: None,
+                            last_modified_at: None,
+                            mutability: metadata_url.as_str().into(),
+                        });
+                        self.metrics.inc_tasks("media", MetricStatus::FAILURE);
+                    },
+                    JsonDownloaderError::CouldNotDeserialize => {
+                        pg_updates.push(UpdatedTask {
+                            status: TaskStatus::Failed,
+                            metadata_url: metadata_url.clone(),
+                            etag: None,
+                            last_modified_at: None,
+                            mutability: metadata_url.as_str().into(),
+                        });
+                        self.metrics.inc_tasks("json", MetricStatus::FAILURE);
+                    },
+                    JsonDownloaderError::CouldNotReadHeader => {
+                        pg_updates.push(UpdatedTask {
+                            status: TaskStatus::Failed,
+                            metadata_url: metadata_url.clone(),
+                            etag: None,
+                            last_modified_at: None,
+                            mutability: metadata_url.as_str().into(),
+                        });
+                        self.metrics.inc_tasks("unknown", MetricStatus::FAILURE);
+                    },
+                    //TODO: should we log the error status code?
+                    JsonDownloaderError::ErrorStatusCode(_err) => {
+                        pg_updates.push(UpdatedTask {
+                            status: TaskStatus::Failed,
+                            metadata_url: metadata_url.clone(),
+                            etag: None,
+                            last_modified_at: None,
+                            mutability: metadata_url.as_str().into(),
+                        });
+
+                        self.metrics.inc_tasks("json", MetricStatus::FAILURE);
+                    },
+                    JsonDownloaderError::ErrorDownloading(_err) => {
+                        // Revert to pending status to retry until max attempts
```

Comment:
Does this comment correspond with reality? We set status to `Failed`. Or I missed something?

Reply:
You're correct, the comment is redundant. Leftovers from the previous version.